lib/fortnox/api/repositories/authentication.rb

```diff
@@ -0,0 +1,78 @@
+require_relative 'base'
+
+module Fortnox
+  module API
+    module Repository
+      class Authentication
+        def renew_access_token(refresh_token)
+          raise ArgumentError, 'Refresh token is empty!' if refresh_token.nil? || refresh_token.empty?
+
+          credentials = Base64.encode64("#{client_id}:#{client_secret}")
+
+          renew_headers = {
+            'Content-type' => 'application/x-www-form-urlencoded',
+            Authorization: "Basic #{credentials}"
+          }
+
+          body = {
+            grant_type: 'refresh_token',
+            refresh_token: refresh_token
+          }
+
+          response = HTTParty.post(config.token_url, headers: renew_headers, body: body)
+
+          handle_errors(response)
+
+          response_to_hash(response.parsed_response)
+        end
+
+        private
+
+        def response_to_hash(parsed_response)
+          {
+            access_token: parsed_response['access_token'],
+            refresh_token: parsed_response['refresh_token'],
+            expires_in: parsed_response['expires_in'],
+            token_type: parsed_response['token_type'],
+            scope: parsed_response['scope']
+          }
+        end
+
+        def handle_errors(response)
+          raise Fortnox::API::RemoteServerError, "Bad request. Error: \"#{response.body}\"" if response.code == 400
+
+          if response.code == 401
```

Comment:
Are we? This repository does not inherit from `Repository::Base` since it's so special, so there's no other place we catch `401` here.

Regarding `403`, how would we get such an error when asking for new tokens?

Reply:
Invalid refresh token => 403.

Ok, but the error handling is the same? But maybe other things would interfere if you inherited from `Repository::Base`?

Reply:
I'll try to send them an invalid token and see what will happen.

Reply:
An invalid refresh token is tested here:

```
      before do
        # VCR: Valid config required
        Fortnox::API.configure do |config|
          config.client_secret = ENV.fetch('FORTNOX_API_CLIENT_SECRET')
          config.client_id = ENV.fetch('FORTNOX_API_CLIENT_ID')
        end
      end

      context 'with invalid refresh token' do
        it 'raises an error' do
          expect do
            VCR.use_cassette("#{vcr_dir}/invalid_refresh_token") do
              repository.renew_tokens('invalid_refresh_token')
            end
          end.to raise_error(Fortnox::API::RemoteServerError, /Bad request(.)*Error:(.)*Invalid refresh token/)
        end
      end
```

it returns
```
...
status: 400
'{"error":"invalid_grant","error_description":"Invalid refresh token"}'
```

Reply:
`Repository::Base` requires an `access_token`, a mapper and `keys_to_filter` etc. We can't use that class here.